### PR TITLE
[DebugInfo] Fix faulty DIExpression::appendToStack assert

### PR DESCRIPTION
**Pull request by dstenb**
The appendToStack() function asserts that no DW_OP_stack_value or
DW_OP_LLVM_fragment operations are present in the operations to be
appended. The function did that by iterating over all elements in the
array rather than just the operations, leading it to falsely asserting
on the following input produced by getExt(), since 159 (0x9f) is the
DWARF code for DW_OP_stack_value:

  {dwarf::DW_OP_LLVM_convert, 159, dwarf::DW_ATE_signed}

Fix this by using expr_op iterators.

### CODE REVIEW

**Review by OCHyams — APPROVED** [review]
Ah, nice find. LGTM, thanks.

File: llvm/test/DebugInfo/Generic/append-to-stack-assert.ll [hunk 1]
```diff
@@ -0,0 +1,49 @@
+; RUN: opt -passes=instcombine -S < %s | FileCheck %s
```
[the remark this answers is not in the document]
Comment (OCHyams) [reply llvm/test/DebugInfo/Generic/append-to-stack-assert.ll:1]: +1, seems reasonable to explore putting this in a unittest instead

[the remark this answers is not in the document]
Comment (dstenb) [reply llvm/test/DebugInfo/Generic/append-to-stack-assert.ll]: Sure! I have changed this to a unit test now.

**Review by felipepiovezan — APPROVED** [review]
LGTM! Thanks for improving the test

**Comment by dstenb** [comment]
Thanks for the reviews!

**Pull request merged**